[Finishes #121007727] show English longform instructions in CSP regardless of locale

Bjvanminnen opened the pull request:
In CSF, we never want to use markdownInstructions in non-English locales as they aren't translated. However, in CSP we do want to use those because we're English only.

What was happening was that if my language was French, for example, we would set longInstructions to be null, then because we didn't have longInstructions we would set longInstructions to be shortInstructions. Fix causes us to use the markdownInstructions regardless of locale in CSP, and I think the code also reads better.

islemaster commented:
LGTM.

This logic keeps changing.  I wonder if we should extract it into a little helper, just to isolate this business logic and make it obvious which parameters go into the decision (and maybe write a unit test).

``` javascript
this.reduxStore.dispatch(
    setInstructionsConstants(
        getInstructionsConstants(
            level.instructions,
            level.markdownInstructions,
            config.locale,
            !!config.noInstructionsWhenCollapsed)));
```

Bjvanminnen commented:
I extracted a helper and put it in instructions.js. I then also added some tests for this. In the course of doing this, I split out instructionsTest.js into instructionsComponentsTests.js and instructionsTest.js (such that instructionsTests.js is only testing instructions.js). Enough delta here that it might be worth another quick look if you have a chance @islemaster

islemaster commented:
:+1: LGTM!
